gcloud/resource_manager/test_project.py

```diff
@@ -0,0 +1,293 @@
+
+
+class TestProject(unittest2.TestCase):
+
+    def _getTargetClass(self):
+        from gcloud.resource_manager.project import Project
+        return Project
+
+    def _makeOne(self, *args, **kw):
+        return self._getTargetClass()(*args, **kw)
+
+    def test_from_api_repr(self):
+        """Projects should be created properly from an API representation."""
+        client = _Client()
+        PROJECT_ID = 'project-id'
+        PROJECT_NAME = 'My Project Name'
+        PROJECT_NUMBER = 12345678
+        PROJECT_LABELS = {'env': 'prod'}
+        PROJECT_LIFECYCLE_STATE = 'ACTIVE'
+        resource = {'projectId': PROJECT_ID,
+                    'name': PROJECT_NAME,
+                    'projectNumber': PROJECT_NUMBER,
+                    'labels': PROJECT_LABELS,
+                    'lifecycleState': PROJECT_LIFECYCLE_STATE}
+        project = self._getTargetClass().from_api_repr(resource, client)
+        self.assertEqual(project.project_id, PROJECT_ID)
+        self.assertEqual(project.name, PROJECT_NAME)
+        self.assertEqual(project.number, PROJECT_NUMBER)
+        self.assertEqual(project.labels, PROJECT_LABELS)
+        self.assertEqual(project.status, PROJECT_LIFECYCLE_STATE)
+
+    def test_constructor(self):
+        """Constructing a project should set properties appropriately."""
+        client = _Client()
+        PROJECT_ID = 'project-id'
+        project = self._makeOne(project_id=PROJECT_ID, client=client)
+        self.assertEqual(project.project_id, PROJECT_ID)
+        self.assertEqual(project.client, client)
+        self.assertEqual(project.name, None)
+        self.assertEqual(project.number, None)
+        self.assertEqual(project.labels, {})
+        self.assertEqual(project.status, None)
+
+    def test_full_name(self):
+        """Check that the full_name property works."""
+        PROJECT_ID = 'project-id'
+        project = self._makeOne(project_id=PROJECT_ID, client=_Client())
+        self.assertEqual('projects/%s' % PROJECT_ID, project.full_name)
+
+    def test_full_name_missing_id(self):
+        """Getting the full_name property with no ID should raise an error."""
+        project = self._makeOne(project_id=None, client=_Client())
+        with self.assertRaises(ValueError):
+            self.assertIsNone(project.full_name)
+
+    def test_path(self):
+        """Check that the path property works."""
+        PROJECT_ID = 'project-id'
+        project = self._makeOne(project_id=PROJECT_ID, client=_Client())
+        self.assertEqual('/projects/%s' % PROJECT_ID, project.path)
+
+    def test_create(self):
+        """Creating an object should make a specific HTTP request."""
+        PROJECT_RESOURCE = {
+            'projectId': 'project-id',
+            'projectNumber': 123,
+            'name': 'Project Name',
+            'labels': {},
+            'lifecycleState': 'ACTIVE'}
+        connection = _Connection(PROJECT_RESOURCE)
+        client = _Client(connection=connection)
```

Comment:
May be able to get away with only a mock `Connection` if you'd like to keep the tests closer to the real code paths. Your call.